### ISSUE TITLE
Can't select only 1 input when inputs are slightly close to each other

### ISSUE THREAD

**Issue by AshKetshup** [ocornut/imgui#7581]
### Version/Branch of Dear ImGui:

Version 1.89.3 , Branch: master

### Back-ends:

imgui_impl_opengl3.cpp + imgui_impl_glfw.cpp

### Compiler, OS:

Windows 11 + MSVC 2022

### Details:

**My Issue/Question:**

So, I've been using ImGui for over 1 year in this project and I've found this problem more than one time.
When I display multiple inputs slightly close to each other it ends up on trying to select only one of them pick everything from that column p.e... I've seen this appear with ungrouped input too. Here's a few lines of code where this problem exhists:

Even tho I'm using spacing and grouping the input by row it does not allow me to only select one input.
Here's a video to showcase the problem:

### Screenshots/Video:

[Screen Recording - Made with RecordCast(2).webm](https://github.com/ocornut/imgui/assets/58540390/df878eb0-3d8b-43b1-83da-b69957a66d06)

### Minimal, Complete and Verifiable Example code:

```cpp
if (ImGui::Begin("Tesselation Shader Properties")) {
	if (tS.levels.size() == 0)
		tS.levels.push_back({ 0.f, 0.f });

	sortVPair(tS.levels);
	static ImGuiTableFlags flags = ImGuiTableFlags_ScrollY | ImGuiTableFlags_RowBg | ImGuiTableFlags_BordersOuter | ImGuiTableFlags_BordersV | ImGuiTableFlags_Reorderable | ImGuiTableFlags_Hideable;

	// When using ScrollX or ScrollY we need to specify a size for our table container!
	// Otherwise by default the table will fit all available space, like a BeginChild() call.
	if (ImGui::BeginTable("table_scrolly", 1, flags, ImVec2(-FLT_MIN, 5 * ImGui::GetTextLineHeightWithSpacing()))) {

		for (size_t row = 0; row < tS.levels.size(); row++) {
			ImGui::BeginGroup();

			ImGui::TableNextRow();

			float* x[2] = { &tS.levels[row].first, &tS.levels[row].second };

			ImGui::TableNextColumn();
			ImGui::DragFloat2("", *x);

			ImGui::EndGroup();

			ImGui::Spacing();
		}

		ImGui::EndTable();

	}
(...)
```

**Comment by GamingMinds-DanielC** [comment]
Classic id conflict, see here for details:
https://github.com/ocornut/imgui/blob/master/docs/FAQ.md#q-about-the-id-stack-system

You basicalls have submitted multiple selectables with the same id, reacting as one. One way you can solve it is by using `ImGui::PushID( (int)row );` and `ImGui::PopID();` at the start and end of your loop body.

**Comment by ocornut** [comment]
Closing as answered, thank you Daniel! 
Indeed the provided link should give you all answers.
I will amend the FAQ entry with a GIF to highlight the common issue.

**Comment by ocornut** [comment]
I added code (c383795e) to generate this gif:
![capture_faq_idstack_gif_0000](https://github.com/ocornut/imgui/assets/8225057/f21ae70e-1f40-4d4c-a18e-aba387b9e9d8)
And added it to the FAQ entry, along with corresponding UI code.

**Comment by AshKetshup** [comment]
Thanks for your time, I've been so lost lately with this problem

**Comment by AshKetshup** [comment]
I'm sorry I have to reopen the issue, I'm not sure if I understood this. I tried using the PushID and PopID calls to manage this it.
I've read somewhere in a closed issue that naming a label with a pre "##" would hide the label during rendering, even tho it is true on a different row of my table ends up pasting the label without the first character. While it starts messing with the other input fields. Not sure if it is a pointer problem or something related to me using a `vector<pair<float, float>> levels` to this particular case. I'll be posting also a gif representing this problem.

![Recording 2024-05-15 at 04 35 46](https://github.com/ocornut/imgui/assets/58540390/17c57bbe-acf5-445a-8f91-6f0f6e0384e9)

Here's my updated code:
```cpp
{
	if (ImGui::Begin("Tesselation Shader Properties")) {
		if (tS.levels.size() == 0)
			tS.levels.push_back({ 0.f, 0.f });

		sortVPair(tS.levels);
		
		static ImGuiTableFlags flags = ImGuiTableFlags_ScrollY | ImGuiTableFlags_RowBg | ImGuiTableFlags_BordersOuter | ImGuiTableFlags_BordersV | ImGuiTableFlags_Reorderable | ImGuiTableFlags_Hideable;

		if (ImGui::BeginTable("table_scrolly", 2, flags, ImVec2(-FLT_MIN, 5 * ImGui::GetTextLineHeightWithSpacing()))) {
			for (size_t row = 0; row < tS.levels.size(); row++) {
				ImGui::TableNextRow();
				{
					ImGui::TableNextColumn();

					ImGui::PushID(row);
					ImGui::DragFloat("##DistanceDrag" + (row), &( tS.levels[row].first ), 1.f, 0.f, 0.f, "Distance > %.1f");
					ImGui::PopID();
				}

				{
					ImGui::TableNextColumn();

					ImGui::PushID(row);
					ImGui::DragFloat("##LevelDrag" + (row), &( tS.levels[row].second ), 1.f, 0.f, 0.f, "Level : %.1f");
					ImGui::PopID();
				}
			}

			ImGui::EndTable();

		}

		if (ImGui::Button("+"))
			tS.levels.push_back({ 0.f, 0.f });

		ImGui::SameLine();

		ImGui::BeginDisabled(tS.levels.size() <= 1);
		if (ImGui::Button("-"))
			tS.levels.pop_back();
		ImGui::EndDisabled();


		ImGui::End();
	}
}
```

**Comment by GamingMinds-DanielC** [comment]
Your problem is here:
`"##DistanceDrag" + (row)`

That is not how you construct strings in C++. That's pointer arithmetic. You basically take the string, which is a `char*`, then offset the pointer itself by the current value of row. So your labels for rows 0, 1 and 2 are `"##DistanceDrag"`, `"#DistanceDrag"` and `"DistanceDrag"` respectively, which fits with the observed behavior.

If you want to build your labels with values pasted in, that's called string formatting. One of many ways: take a look at `sprintf_s` and use it to format a string into a local char array of sufficient size.

**Comment by AshKetshup** [comment]
@GamingMinds-DanielC That does solve the label issue, for sure. I was wondering if it actually had something to do to pointer arithmetic. Although the behaviour itself of the change of values affecting the other fields keeps going and since the smoking gun of pointer arithmetic was already found, i then don't know what might be the problem in the first place. Nice catch tho, thanks.

TLDR: Labels are now actually invisible but the fields interfering with each other persists.

**Comment by GamingMinds-DanielC** [comment]
> TLDR: Labels are now actually invisible but the fields interfering with each other persists.

I'm pretty sure this is your culprit: `sortVPair(tS.levels);`

You should avoid changing underlying values (if you swap them due to sorting, that counts) while you are actively editing them. My advice would be to sort only after an edit, `ImGui::IsItemDeactivatedAfterEdit()` can help with that.

**Comment by AshKetshup** [comment]
Thanks, that seems to solve the case! You were right. Since I have your attention do you know how in my case I could make the element widgets stretch themselves to the full width of the cell?

**Comment by GamingMinds-DanielC** [comment]
> Since I have your attention do you know how in my case I could make the element widgets stretch themselves to the full width of the cell?

Never tried if that works in table cells, but you can try `ImGui::SetNextItemWidth(-FLT_MIN)`. Slightly negative as an offset to the available size. If that doesn't work, you need to calculate or know the size yourself and can use that as a positive value.

**Comment by ocornut** [comment]
FYI if you use `PushID(row)` you don't need to decorate label names, a simple `##DistanceDrag` works.

Please browse the Tables section of the Demo as it exhibit many use cases.

**Comment by AshKetshup** [comment]
Thanks to you both. Its all up and running. Thanks for your help!
